### ISSUE TITLE
Vite HMR update is conflicting with Windi JIT?

### ISSUE THREAD

**Issue by Yofou** [windicss/vite-plugin-windicss#238]
**Describe the bug**

Windi HMR/JIT doesn't generate new Utility classes on demand (after the first load)

**Minimal reproduction**
**ensure you're on the v2.5.4 on vite** after that run up the sever and make changes to your elements
(ie add/modify new utils), you can see an example of this here on the [WindiCSS Discord](https://discord.com/channels/810981250981953597/816540947860684871/890010348915150888)

**temporary fix**
downgrade to `vite@2.5.3` and everything works fine and dandy again

**Versions**
- vite-plugin-windicss: `1.4.4`
- vite: `2.5.4`
- framework(vue/react/svelte/etc): `svelte-(ts)`

**Comment by dominikg** [comment]
sounds like this change could play a role https://github.com/vitejs/vite/commit/b3b8c61b37f36713605d61baeb98cfc0deb0b020

**Comment by dominikg** [comment]
The change i linked above was released with vite 2.5.5, and in my local testing 2.5.4 still works, sending an hmr update for both the svelte component and windi.css which contains the new utility class.

steps:

```bash
pnpx svite create --t windicss windi-css-update-missing-repro
cd windi-css-update-missing-repro
pnpm i
pnpm i -D vite@2.5.4 @sveltejs/vite-plugin-svelte@1.0.0-next.25
pnpm dev
# edit src/App.svelte, change m-2 in line 6 to m-10. 
# Notice that 2 hmr updates happen, one for App.svelte and another one for windi.css, containing the new m-10 utility
# stop dev server with ctrl-c
pnpm i -D vite@2.5.5
pnpm dev
# edit src/App.svelte, change m-10 in line 6 to m-20. 
# Notice that only one hmr update happens for App.svelte and the h1 now does not have a margin applied
```

**Comment by dominikg** [comment]
I removed the `break` added in https://github.com/vitejs/vite/commit/b3b8c61b37f36713605d61baeb98cfc0deb0b020  and it works again with vite@2.7.0-beta.0 and @sveltejs/vite-plugin-svelte@1.0.0-next.30  

The problem here is that a single file save of a .svelte file triggers 2 different hmr updates by 2 different plugins which are collaborating via plugin api. It looks like this case wasn't considered when that break was added to vite as an optimization.

I see 3 possible fixes:
1) revert the change in vite, allowing multiple plugins to send updates from a single file change event again
2) change the implementation in vite-plugin-windicss to use a svelte markup preprocessor to scan the file content  and trigger hmr update for the virtual windi.css module from the preprocessor
3) vite-plugin-svelte somehow triggers a separate asnyc update event for windi.css 

@antfu what do you think?

**Comment by antfu** [comment]
If the hmr events were handled by Svelte first and caused Windi not to receive the event, I guess we should revert it. If it's another way around, I think I can do some trick on Windi side to not return the modules and let it pass to Svelte.

**Comment by dominikg** [comment]
svelte is currently used first, though that could be changed (depends on plugin order in user config). It could work if windi was first and didn't return a change from handleHotUpdate, but that feels rather hacky to me.

**Comment by dominikg** [comment]
This was fixed in vite 2.6.14 

https://github.com/vitejs/vite/blob/release/2.6.x/packages/vite/CHANGELOG.md#2614-2021-11-09
